src/reporters/console/console-reporter.js

```diff
@@ -186,42 +232,37 @@ export default class ConsoleReporter extends BaseReporter {
       });
     });
   }
-
+  // handles basic tree output to console
   tree(key: string, trees: Trees) {
-    trees = sortTrees(trees);
-
-    const stdout = this.stdout;
-
+    // 
     const output = ({name, children, hint, color}, level, end) => {
-      children = sortTrees(children);
-
-      let indent = end ? '└' : '├';
-
-      if (level) {
-        indent = repeat('│  ', level) + indent;
-      }
-
-      let suffix = '';
-      if (hint) {
-        suffix += ` (${this.format.grey(hint)})`;
-      }
-      if (color) {
-        name = this.format[color](name);
-      }
-      stdout.write(`${indent}─ ${name}${suffix}\n`);
+      this.writeOutput(this.getFormattedOutput(end, level, hint, color, name, 0));
+      (children && children.length) && recurseTree(sortTrees(children), level, output);
+    };
+    // let recursive call handle giving output level 0 
+    recurseTree(sortTrees(trees), -1, output);
+  }
 
-      if (children && children.length) {
-        for (let i = 0; i < children.length; i++) {
-          const tree = children[i];
-          output(tree, level + 1, i === children.length - 1);
-        }
+   // handles detailed tree output to console
+  extendedTree(key: string, trees: Trees) {
+    // 
+    const output = ({name, type, registry, reference, children, hint, color}, level, end) => {
+      const _this = this;
```

Comment:
This adds support for `yarn ls --long` or `yarn ls --la`, which is a nod to [npm's](https://docs.npmjs.com/cli/ls#long) `npm ls --long`

The two paths of outputting a tree diverge enough that I needed to break the [original tree ](https://github.com/yarnpkg/yarn/blob/master/src/reporters/console/console-reporter.js#L190) function into smaller bits that could be consumed by another function, `extendedTree` you see here